tools/lsp/server_loop.rs

```diff
@@ -235,6 +271,10 @@ impl OffsetToPositionMapper {
     pub fn map_range(&self, r: rowan::TextRange) -> lsp_types::Range {
         lsp_types::Range::new(self.map(r.start()), self.map(r.end()))
     }
+
+    pub fn map_node(&self, n: &SyntaxNode) -> lsp_types::Range {
```

Comment:
could you add documentation saying why this is different than `map_range(n.text_range())`?

I think what you're doing is trimming spaces at the beginning and the end of it.

Note that ideally, the parser shouldn't put the space in the node if the space doesn't belong there. That should also help for the diagnostics.  But i understand this might not be something you want to look at now.